File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/userCriteria/environmentalParameterUserCriteria/EnvParamDateUserCriteria.ts
import {AbstractEnvParamUserCriteria} from './AbstractEnvParamUserCriteria';

export class EnvParamDateUserCriteria extends AbstractEnvParamUserCriteria {
  minDate: Date = new Date();
  maxDate: Date = new Date();
  date: Date = new Date();
}

File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/userCriteria/environmentalParameterUserCriteria/EnvParamDateUserCriteria.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.userCriteria.environmentalParameterUserCriteria;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.Instant;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor

@DiscriminatorValue("DATE")
public class EnvParamDateUserCriteria extends AbstractEnvParamUserCriteria {
    private Instant minDate;
    private Instant maxDate;
    private Instant date;
}
